Basic_RBC6.jl

```diff
@@ -1,106 +1,141 @@
+                           f_policy::Matrix,
+                           E::Matrix)
+    
+    α,β,k_grid,z =_unpack(p)
+    n = length(k_grid)
+    m = length(z)    
+    #-----------------------------------------------------------------
+    for i = 1:m
+    # We start from previous choice (monotonicity of policy function)
+        kNext = 1
+        for j = 1:n
+            v_max = -1000.0
+            k_choice  = k_grid[1]
+                for l = kNext : n
+                    c = output[j,i]-k_grid[l]
+                    v = (1-β)*log(c)+β*E[l,i]
+                    if v > v_max
+                        v_max = v
+                        k_choice = k_grid[l]
+                        kNext = l
+                    else
                         break # We break when we have achieved the max
-                        end
-                                 
                     end
-            
-                    mValueFunctionNew[nCapital,nProductivity] = valueHighSoFar
-                    mPolicyFunction[nCapital,nProductivity] = capitalChoice
-          
+                                 
                 end
-        
-            end
+                f_valueN[j,i] = v_max
+                f_policy[j,i] = k_choice
+          end
+    end
+    #-----------------------------------------------------------------
+    
+end
 
-            maxDifference  = maximum(abs.(mValueFunctionNew-mValueFunction))
-            mValueFunction    = mValueFunctionNew
-            mValueFunctionNew = zeros(nGridCapital,nGridProductivity)
 
-            iteration = iteration+1
-            if mod(iteration,10)==0 || iteration == 1
-                println(" Iteration = ", iteration, " Sup Diff = ", maxDifference)
-            end
-           
-        end
+#====================================#
+# m : length of GridProductivity(z)
+# n : length of GridCapital(k_grid)
+# tol : tolerance
+# E : expectedValueFunction
+# f_value : mValueFunction
+# f_valueN : mValueFunctionNew
+# f_policy : mPolicyFunction
+# error : maxDifference
+#iter : iteration
+#====================================#
 
-        println(" Iteration = ", iteration, " Sup Diff = ", maxDifference)
+function main()
+    
+        p=BasicRBC()
+        α,β,k_grid,z,T = _unpack(p)
+        n = length(k_grid)
+        m = length(z)
+        #-----------------------------------------------------------------
+        ouput    = zeros(n,m)
+        f_value  = zeros(n,m)
+        f_valueN = zeros(n,m)
+        f_policy = zeros(n,m)
+        E = zeros(n,m)
+        output = (k_grid.^α)*z;
+        #-----------------------------------------------------------------
+        error = 10.0
+        tol = 0.0000001
+        iter = 0
+        #-----------------------------------------------------------------
+        while error > tol
+            E = f_value*T';
+            bellman_operator!(p ,Matrix(output),Matrix(f_value),Matrix(f_valueN),Matrix(f_policy),Matrix(E))
```

Comment:
Thank you, I followed the instruction from "Handle excess argument diversity in the caller" in this tutorial https://docs.julialang.org/en/stable/manual/style-guide/